redbot/core/settings_caches.py

```diff
@@ -214,15 +218,17 @@ def __init__(self, config: Config):
             isinstance(r_or_u, str) for r_or_u in role_or_user
         ):
             raise TypeError("Blacklisted objects must be a list of ints")
+        if gid not in self._cached_blacklist:
+            self._cached_blacklist[gid] = []
```

Comment:
This `if` may result in adding empty list to the cache when in fact there is something in current blacklist which just wasn't cached just yet.